### PR TITLE
Support for document literals.

### PR DESCRIPTION
**Pull request by alecthomas**
This allows JSON documents to be expressed in the parser rather than
passing opaque JSON blobs through.

### CODE REVIEW

**Comment by alecthomas** [comment]
I can wire this up to INSERT in a followup PR if you like the approach. I won't be offended if you don't :)

File: parser/parser.go [hunk 1]
```diff
@@ -284,13 +284,44 @@ func (p PathFragment) String() string {
 	return buf.String()
 }
 
+type ObjectEntry struct {
+	Key   string `@(Ident | QuotedIdent | String)`
```
Comment (mightyguava) [inline parser/parser.go:288]: Can we keep to strict json? Supporting ident allows you to omit quotes and quotedident would be backing quotes

File: parser/parser.go [hunk 2]
```diff
@@ -284,13 +284,44 @@ func (p PathFragment) String() string {
...
+	Value *Value `":" @@`
+}
+
+type Object struct {
+	Entries []*ObjectEntry `"{" (@@ ("," @@)* ","?)? "}"`
```
Comment (mightyguava) [inline parser/parser.go:293]: This allows trailing comma?

**Review by mightyguava — APPROVED** [review]

File: parser/parser.go (hunk 2 above)
Comment (alecthomas) [reply parser/parser.go:293]: Rectifying the GREAT MISTAKE in JSON!

**Pull request merged**